sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingXPathBase.cs

```diff
@@ -0,0 +1,59 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class ExecutingXPathBase<TSyntaxKind> : SonarDiagnosticAnalyzer
+        where TSyntaxKind : struct
+    {
+        protected const string DiagnosticId = "S4817";
+        protected const string MessageFormat = "Make sure that executing this XPATH expression is safe.";
+
+        protected abstract DiagnosticDescriptor Rule { get; }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(Rule);
+
+        protected abstract InvocationTracker<TSyntaxKind> Tracker { get; }
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            Tracker.Track(context,
+
+                Tracker.MatchSimpleNames(
```

Comment:
[cosmetic] I would prefer `InvocationTracker` instead of `Tracker` as property name. Since the *Tracker methods are relatively simple, it is not so easy (for me) to guess which exact kind of tracker this is...

If you are fixing this, please, remove the empty line at line 43 :)